libsolidity/interface/StandardCompiler.cpp

```diff
@@ -657,72 +657,82 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 
 	ret.errors = Json::arrayValue;
 
-	for (auto const& sourceName: sources.getMemberNames())
+	if (ret.language == "Solidity" || ret.language == "Yul")
 	{
-		string hash;
+		for (auto const& sourceName: sources.getMemberNames())
+		{
+			string hash;
 
-		if (auto result = checkSourceKeys(sources[sourceName], sourceName))
-			return *result;
+			if (auto result = checkSourceKeys(sources[sourceName], sourceName))
+				return *result;
 
-		if (sources[sourceName]["keccak256"].isString())
-			hash = sources[sourceName]["keccak256"].asString();
+			if (sources[sourceName]["keccak256"].isString())
+				hash = sources[sourceName]["keccak256"].asString();
 
-		if (sources[sourceName]["content"].isString())
-		{
-			string content = sources[sourceName]["content"].asString();
-			if (!hash.empty() && !hashMatchesContent(hash, content))
-				ret.errors.append(formatError(
-					Error::Type::IOError,
-					"general",
-					"Mismatch between content and supplied hash for \"" + sourceName + "\""
-				));
-			else
-				ret.sources[sourceName] = content;
-		}
-		else if (sources[sourceName]["urls"].isArray())
-		{
-			if (!m_readFile)
-				return formatFatalError(Error::Type::JSONError, "No import callback supplied, but URL is requested.");
+			if (sources[sourceName]["content"].isString())
+			{
+				string content = sources[sourceName]["content"].asString();
+				if (!hash.empty() && !hashMatchesContent(hash, content))
+					ret.errors.append(formatError(
+						Error::Type::IOError,
+						"general",
+						"Mismatch between content and supplied hash for \"" + sourceName + "\""));
```

Comment:
This and a few of the other changes here are actually coding style violations that snuck in here somehow.